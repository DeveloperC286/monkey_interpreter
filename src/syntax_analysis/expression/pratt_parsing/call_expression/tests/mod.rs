use rstest::rstest;

#[rstest(
    code,
    snapshot_name,
    case("add(5, 12)", "test_syntax_analysis_for_call_expression_case1"),
    case("add(3, 7 * 2);", "test_syntax_analysis_for_call_expression_case2")
)]
fn test_syntax_analysis_for_call_expression(code: &str, snapshot_name: &str) {
    assert_expected_returned_abstract_syntax_tree!(code, snapshot_name);
}
