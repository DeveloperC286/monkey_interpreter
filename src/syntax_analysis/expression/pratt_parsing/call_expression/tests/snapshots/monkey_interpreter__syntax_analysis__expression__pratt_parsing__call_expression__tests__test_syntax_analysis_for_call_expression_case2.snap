---
source: src/syntax_analysis/expression/pratt_parsing/call_expression/tests/mod.rs
expression: "crate::syntax_analysis::SyntaxAnalysis::from(crate::lexical_analysis::LexicalAnalysis::from(code).unwrap())"

---
Ok(
    [
        Expression {
            expression: Call {
                function: Identifier {
                    identifier: "add",
                },
                arguments: [
                    Integer {
                        literal: 3,
                    },
                    Infix {
                        left_hand: Integer {
                            literal: 7,
                        },
                        operator: Multiply,
                        right_hand: Integer {
                            literal: 2,
                        },
                    },
                ],
            },
        },
    ],
)
